NazarankaYuliya/REACT2023Q4
# Week3

1. Task: [link](https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module03)
2. Screenshot:
<img width="706" alt="Screenshot 2023-11-12 at 23 06 24" src="https://github.com/NazarankaYuliya/REACT2023Q4/assets/97028438/628d7b38-3b6a-472e-b98c-7ff3afb1dec8">

3. Deploy: 
4. Done 12.11.2023 / deadline 13.11.2023
5. Score: 25 / 100
  - [x] Custom state management using the Context API is implemented - 5 points
  - [x] The Search component value is stored in the context - 5 points
  - [x] The list of items received from the API is stored in the context - 5 points
  - [x] The context is used in components that need access to the data - 5 points
  - [x] React Testing Library and Jest or Vitest are added and configured - 10 points
  - [ ] Test cases - 15 points (5 points per each):
  - Check that a loading indicator is displayed while fetching data;
  - Ensure that clicking the close button hides the component.
  - Ensure that the 404 page is displayed when navigating to an invalid route.
  - [x] Husky runs tests on pre-push - 10 points


Penalties:
 - [x]Test coverage is less than 80%: -30 points
